### DISCUSSION TITLE
Device Offline

### DISCUSSION THREAD

**Discussion by arrmo** [milo526/homebridge-tuya-web#363] [General]
Hi,

OK, this may be me - let me say that up front 😆. I have several Tuya devices, all working OK, except this one keeps having issues, getting this error message in the HomeBridge log (not 100% of the time, but quite often),
```
[TuyaWebPlatform] [emRussell] - Device Offline
```

I have checked the device in Smart Life - NP, it's online and reporting (Energy Management data even). And of course then I exit Smart Life. I also check it with (Python) via tinytuya, and it's OK - about 98.5% of the time (I run a regular check on it, ~ 1.5% of the time it fails). I have other devices, even the same model and manufacturer, with no issues.

Any thoughts or suggestions, things to try? I have power cycled it (unplugged, plugged back in), no difference. BTW, this is the [device](https://www.amazon.com/gp/product/B07CVPKD8Z/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1).

Any thoughts would be greatly appreciated. Thanks!

**Comment by milo526** [comment]
Please post the result of [the first two steps under "How to check whether the API this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device), perform this check when the device is misbehaving. 

I'm afraid that the problem here is that I can only query the device status once every 5 minutes. Whenever the device does not respond it will keep this status for 5 minutes. It might be the case that this leads to you noticing it more often instead of it happening more often.

**Comment by arrmo** [comment]
Sure, NP! Here is the output, for this device. I removed the id for now, thinking I should?
```
{'data': {'online': False, 'state': True},
'dev_type': 'switch',
'ha_type': 'switch',
'icon': 'https://images.tuyaus.com/smart/icon/ay1518058362399MS7Ky/1584504180e1117f81cd4.png',
'id': 'xxxxx',
'name': 'emRussell'}
```

So then I opened the SmartLife app ... it shows Offline, but almost immediately changes back to online - and working. Something in the Tuya library is perhaps causing this?

Thanks!
